app/services/service_provider_config.rb

```diff
@@ -0,0 +1,22 @@
+class ServiceProviderConfig
+  def initialize(filename:, issuer:)
+    @issuer = issuer
+    @data = YAML.load_file("#{Rails.root}/config/#{filename}")
```

Comment:
Does this code run more than once per request?   Even if its only once per request wouldn't it be better not to load and parse this for every request?